src/main/java/org/rumbledb/server/RumbleHandler.java

```diff
@@ -0,0 +1,203 @@
+package org.rumbledb.server;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URLDecoder;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.spark.SparkException;
+import org.rumbledb.api.Item;
+import org.rumbledb.cli.JsoniqQueryExecutor;
+import org.rumbledb.config.RumbleRuntimeConfiguration;
+import org.rumbledb.exceptions.CliException;
+import org.rumbledb.exceptions.OurBadException;
+import org.rumbledb.exceptions.RumbleException;
+import org.rumbledb.items.ItemFactory;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+
+import sparksoniq.spark.SparkSessionManager;
+
+@SuppressWarnings("restriction")
+public class RumbleHandler implements HttpHandler {
+
+    private enum StatusCode {
+        SUCCESS(200),
+        SERVER_ERROR(500);
+
+        private int code;
+
+        StatusCode(int code) {
+            this.code = code;
+        }
+
+        public int getCode() {
+            return this.code;
+        }
+    }
+
+    public RumbleHandler() {
+    }
+
+    private void sendResponse(HttpExchange exchange, StatusCode code, String response) throws IOException {
+        exchange.sendResponseHeaders(code.getCode(), response.getBytes().length);
+        OutputStream stream = exchange.getResponseBody();
+        stream.write(response.getBytes());
+        stream.close();
+    }
+
+    public String[] getCLIArguments(String query) throws UnsupportedEncodingException {
+        Map<String, String> queryParameters = new HashMap<String, String>();
+        if (query == null) {
+            query = "";
+        }
+        for (String pair : query.split("&")) {
+            int index = pair.indexOf("=");
+            if (index != -1) {
+                queryParameters.put(
+                    URLDecoder.decode(pair.substring(0, index), "UTF-8"),
+                    URLDecoder.decode(pair.substring(index + 1), "UTF-8")
+                );
+            }
+        }
+        String[] args = new String[queryParameters.keySet().size() * 2];
+        int i = 0;
+        for (String param : queryParameters.keySet()) {
+            args[i++] = "--" + param;
+            args[i++] = queryParameters.get(param);
+        }
+        return args;
+    }
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        try {
+            URI uri = exchange.getRequestURI();
+            String queryString = uri.getQuery();
+            String args[] = getCLIArguments(queryString);
```

Comment:
IntelliJ just gave me a heads up that `String args[]` is a C-Style array declaration.

Java Style is as follows: `String[] args`

Reply:
Good catch, done